### PR TITLE
Tests for components

### PR DESCRIPTION
**Pull request by rayto22**
Tests for components 
1. ShelterList
2. ShelterCard
3. ShelterCardDetails

Added interface for auctionLot get request.

### CODE REVIEW

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts [hunk 1]
```diff
@@ -1,24 +1,162 @@
...
+        adressID: component.shelter.adressID,
+        avatar: component.shelter.avatar,
+        locationID: component.shelter.locationID
+      };
+
+      let addressChange = { ...component.profileForm.get('address').value };
+      const isChangeAddress = Object.entries(addressChange).every(([key, value]) => component.shelter.address[key] === value);
+      if (isChangeAddress) {
+        addressChange = null;
+      }
```
Comment (ksenya-ta) [inline src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: do not add any logic to a test, it should have defined inputs/test data without any conditions or uncertain states (and nothing should influence the test run, all the tests should be isolated

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,162 @@
...
+
+      const changeData = {
+        id: component.shelter.id,
+        addressID: component.shelter.adressID,
+        address: addressChange,
+        shelter: shelterChange,
+      };
+
+      component.onSubmit();
+      expect(component[shelterServiceName].putShelterDetails).toHaveBeenCalledWith(changeData);
```
Comment (ksenya-ta) [inline src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: do not use `component[shelterServiceName]` use `sheltersService` instead

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,162 @@
...
+    });
+  });
+
+  describe('Method onEdit', () => {
+    it('should toggle form', () => {
+      const toggleFormMethodName = 'toggleForm';
+      spyOn(component as any, toggleFormMethodName);
+
+      component.onEdit();
+      expect(component[toggleFormMethodName]).toHaveBeenCalled();
```
Comment (ksenya-ta) [inline src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: why `component[toggleFormMethodName]`? do not create additional variables, `toggleForm` should be defined on the component and you can use it directly. if it is private method and you need to test it separately you can access it by `(component as any).toggleForm`

Comment (ksenya-ta) [reply src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: and the same comment for all the other tests

File: src/app/shelters/shelter-list/shelter-list.component.spec.ts [hunk 4]
```diff
@@ -1,34 +1,73 @@
...
+
+  describe('Method onSearch', () => {
+
+    it('sends search params to sheltersService', () => {
+      const searchValue = 'searchValue';
+      spyOn(sheltersService, 'getShelters').and.returnValue(of(mockShelters));
+
+      component.onSearch(searchValue);
+      component.shelters$.subscribe((shelter: Shelter[]) => {
+        expect(shelter).toEqual(mockShelters);
```
Comment (ksenya-ta) [inline src/app/shelters/shelter-list/shelter-list.component.spec.ts]: use done function for async tests

Comment (rayto22) [reply src/app/shelters/shelter-list/shelter-list.component.spec.ts]: Done)

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts (hunk 3 above)
Comment (rayto22) [reply src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: Got it.

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts (hunk 2 above)
Comment (rayto22) [reply src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: Fixed.

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts (hunk 1 above)
Comment (rayto22) [reply src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: logic was removed

File: src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts [hunk 5]
```diff
@@ -1,24 +1,149 @@
...
+        rating: component.shelter.rating,
+        adressID: component.shelter.adressID,
+        avatar: component.shelter.avatar,
+        locationID: component.shelter.locationID
+      };
+      const changeData = {
+        id: component.shelter.id,
+        addressID: component.shelter.adressID,
+        address: null,
+        shelter: shelterChange,
```
Comment (ksenya-ta) [inline src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: btw no need to create separate variable for shelter sub object, remove trailing comma

Comment (rayto22) [reply src/app/shelters/shelter-card-details/shelter-card-details.component.spec.ts]: Will do, thanks

**Pull request merged**